File: Data Structures & Algorithms/maximum-frequency-stack/submission-1.cpp
class FreqStack {
public:
    unordered_map<int , int>freq;
    unordered_map<int , stack<int>>group;
    int maxfreq;
    FreqStack() {
        maxfreq = 0;
        
    }
    
    void push(int val) {
        freq[val]++;
        int f = freq[val];
        maxfreq = max(maxfreq , f);
        group[f].push(val);

        
    }
    
    int pop() {
        int val = group[maxfreq].top();
        group[maxfreq].pop();
        freq[val]--;
        if(group[maxfreq].empty()){
            maxfreq--;
        }
        return val;
        
    }
};

/**
 * Your FreqStack object will be instantiated and called as such:
 * FreqStack* obj = new FreqStack();
 * obj->push(val);
 * int param_2 = obj->pop();
 */

File: Data Structures & Algorithms/online-stock-span/submission-1.java
class StockSpanner {
    ArrayList<Integer>arr = new ArrayList<>();
    Stack<Integer>st;
    public StockSpanner() {
        st = new Stack<>();
        
    }
    
    public int next(int price) {
        arr.add(price);
        while(!st.isEmpty() && arr.get(st.peek())<=arr.get(arr.size()-1)){
            st.pop();
        }
        int ans = 1;
        if(!st.isEmpty()){
        ans = arr.size()-st.peek()-1;
        }else{
            ans = arr.size();
        }
        st.push(arr.size()-1);
        return ans;
        
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */